vyper/compiler/README.md

```diff
@@ -0,0 +1,95 @@
+# 🐍 `vyper.compiler` 🐍
+
+## Purpose
+
+The `vyper.compiler` module contains the main user-facing functionality used to compile
+vyper source code and generate various compiler outputs.
+
+## Organization
+
+`vyper.compiler` has the following structure:
+
+* [`__init__.py`](__init__.py): Contains the `compile_codes` function, which is the
+primary function used for compiling Vyper source code.
+* [`data.py`](data.py): Pure functions for executing each compiler phase, as well as
+the `CompilerData` object that fetches and stores compiler output for each phase.
+* [`output.py`](output.py): Functions that convert compiler data into the final
+formats to be outputted to the user.
+* [`utils.py`](utils.py): Various utility functions related to compilation.
+
+## Control Flow
+
+### Compiler Phases
+
+The compilation process has five broad phases:
+
+1. In [`vyper.ast`](`../ast`), the source code is parsed and converted to an
+abstract syntax tree.
```

Comment:
I sort of want to talk about "AST nodes" briefly here (aka subclasses of `VyperNode`)